### PR TITLE
[feature] [null support # 9] Support null in time related transform functions.

### PR DESCRIPTION
**Pull request by 61yao**
For time related functions, we return null only if the input column argument to the function is null.

See the github issue:
https://github.com/apache/pinot/issues/10252

Design doc:
https://docs.google.com/document/d/1w2eBuZrS73xE2JIw8MYlySy5pNaX9iE8kZr_YAL20CM/edit

### CODE REVIEW

File: pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/CoalesceTransformFunction.java [hunk 1]
```diff
@@ -399,4 +390,22 @@ public String[] transformToStringValuesSV(ValueBlock valueBlock) {
...
+  @Override
+  public RoaringBitmap getNullBitmap(ValueBlock valueBlock) {
+    RoaringBitmap[] nullBitMaps = getNullBitMaps(valueBlock, _transformFunctions);
+    RoaringBitmap bitmap = nullBitMaps[0];
+    for (int i = 1; i < nullBitMaps.length; i++) {
+      RoaringBitmap curBitmap = nullBitMaps[i];
+      if (bitmap != null && curBitmap != null) {
+        bitmap.and(curBitmap);
+      } else {
+        bitmap = null;
```
Comment (KKcorps) [inline pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/CoalesceTransformFunction.java:403]: We can break the loop here

File: pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/DateTimeConversionTransformFunction.java [hunk 2]
```diff
@@ -106,8 +107,8 @@ public void init(List<TransformFunction> arguments, Map<String, ColumnContext> c
 
     _dateTimeTransformer = DateTimeTransformerFactory.getDateTimeTransformer(
         ((LiteralTransformFunction) arguments.get(1)).getStringLiteral(),
-            ((LiteralTransformFunction) arguments.get(2)).getStringLiteral(),
-            ((LiteralTransformFunction) arguments.get(3)).getStringLiteral());
+        ((LiteralTransformFunction) arguments.get(2)).getStringLiteral(),
```
Comment (KKcorps) [inline pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/DateTimeConversionTransformFunction.java:110]: unnecessary formatting

File: pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/GreatestTransformFunction.java [hunk 3]
```diff
@@ -44,6 +47,35 @@ public int[] transformToIntValuesSV(ValueBlock valueBlock) {
...
+    Pair<int[], RoaringBitmap> values = _arguments.get(0).transformToIntValuesSVWithNull(valueBlock);
+    System.arraycopy(values.getLeft(), 0, _intValuesSV, 0, numDocs);
+    RoaringBitmap nullBitmap = values.getRight();
+    for (int i = 1; i < _arguments.size(); i++) {
+      values = _arguments.get(i).transformToIntValuesSVWithNull(valueBlock);
+      RoaringBitmap curNull = values.getRight();
+      for (int j = 0; j < numDocs & j < values.getLeft().length; j++) {
+        // Ignore null values.
+        if (curNull == null || !curNull.contains(j)) {
+          if (nullBitmap != null && nullBitmap.contains(j)) {
```
Comment (KKcorps) [inline pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/GreatestTransformFunction.java:63]: Rename these variables better, curNull and nullBitmap sound confusing. Maybe `combinedNullBitmap` and `currentNullBitmap`

**Review by shenyu0127 — APPROVED** [review]
Overall looks good to me.

File: pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/GreatestTransformFunction.java (hunk 3 above)
Comment (61yao) [reply pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/GreatestTransformFunction.java:63]: Oops.. This doesn't belong to this PR...

File: pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/GreatestTransformFunction.java [hunk 4]
```diff
@@ -44,6 +47,35 @@ public int[] transformToIntValuesSV(ValueBlock valueBlock) {
...
+            _intValuesSV[j] = values.getLeft()[j];
+          } else {
+            _intValuesSV[j] = Math.max(_intValuesSV[j], values.getLeft()[j]);
+          }
+        }
+      }
```
[the remark this answers is not in the document]
Comment (61yao) [reply pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/GreatestTransformFunction.java:69]: This doesn't belong to this PR either... let me rebase

File: pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/CoalesceTransformFunction.java (hunk 1 above)
Comment (61yao) [reply pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/CoalesceTransformFunction.java:403]: This doesn't belong to this PR either..

File: pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/DateTimeConversionTransformFunction.java (hunk 2 above)
Comment (61yao) [reply pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/DateTimeConversionTransformFunction.java:109]: Current formatting is better. so let's keep it?

**Review by shenyu0127 — APPROVED** [review]
LGTM

**Review by KKcorps — APPROVED** [review]
LGTM!

**Review by gortiz — APPROVED** [review]
